pkgs/top-level/php-packages.nix

```diff
@@ -540,7 +540,15 @@ lib.makeScope pkgs.newScope (self: with self; {
             ++ lib.optionals (lib.versionOlder php.version "7.4") [ "--with-libxml-dir=${libxml2.dev}" ];
           doCheck = false;
         }
-        { name = "sockets"; doCheck = false; }
+        {
+          name = "sockets";
+          doCheck = false;
+          patches = lib.optional (php.version == "8.1.2")
+            (fetchpatch {
+              url = "https://patch-diff.githubusercontent.com/raw/php/php-src/pull/7981.patch";
```

Comment:
I would instead point at https://github.com/php/php-src/commit/07aaa34cd418c44f7bc653fafbf49f07fc71b2bf.patch which is the commit that actually made it into their `master`. You may need to use `excludes = [ "NEWS" ];` too as that addition will likely cause a conflict.

Generally we try and avoid pointing at PRs as they can change their contents. Commits that have made it into their `master` are also seen as more "legit" or "the accepted solution" than some PR from some random person.

Reply:
PR has been merged, should we take that in account too ?

Reply:
Well, the PR wasn't actually merged. It was superseded by their own commit. Technically someone could re-open that PR and push something else to that branch. It's also not possible to _see_ that the change is essentially the same without going and looking at that PR and comparing the two changes.. whereas it's very easy to take the `.patch` off the link I provided, look at the commit, see it's in `master` and be happy.

Reply:
True.

However, there is a SHA on the patch, if it changes, we'll get noticed instantely.

But I'm going to follow your advice and update the patch with something better.

Reply:
Yes, I'm not particularly worried about it being _maliciously_ hijacked - it'll just be annoying and confusing if it changes. So that and preferring the merged upstream solution in general makes people wary of seeing `/pull/1234.patch` links.

Reply:
Yup, you,re perfectly right, lesson learned. I will update accordingly.